examples_notebooks/community_contrib/couchbase/GraphRAG_with_Couchbase.ipynb

```diff
@@ -0,0 +1,557 @@
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"RELATIONSHIP_TABLE file not found. Setting relationships to None.\")\n",
+    "    data[\"relationships\"] = None\n",
+    "\n",
+    "try:\n",
+    "    data[\"covariates\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['COVARIATE_TABLE']}.parquet\")\n",
+    "    data[\"covariates\"] = read_indexer_covariates(data[\"covariates\"])\n",
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"COVARIATE_TABLE file not found. Setting covariates to None.\")\n",
+    "    data[\"covariates\"] = None\n",
+    "\n",
+    "try:\n",
+    "    data[\"reports\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['COMMUNITY_REPORT_TABLE']}.parquet\")\n",
+    "    entity_data = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['ENTITY_TABLE']}.parquet\")\n",
+    "    data[\"reports\"] = read_indexer_reports(data[\"reports\"], entity_data, COMMUNITY_LEVEL)\n",
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"COMMUNITY_REPORT_TABLE file not found. Setting reports to None.\")\n",
+    "    data[\"reports\"] = None\n",
+    "\n",
+    "try:\n",
+    "    data[\"text_units\"] = pd.read_parquet(f\"{INPUT_DIR}/{TABLE_NAMES['TEXT_UNIT_TABLE']}.parquet\")\n",
+    "    data[\"text_units\"] = read_indexer_text_units(data[\"text_units\"])\n",
+    "except FileNotFoundError:\n",
+    "    logger.warning(\"TEXT_UNIT_TABLE file not found. Setting text_units to None.\")\n",
+    "    data[\"text_units\"] = None\n",
+    "\n",
+    "logger.info(\"Data loading completed\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "L8AOUrIAgc-8"
+   },
+   "source": [
+    "# Setting Up the Couchbase Vector Store\n",
+    "Couchbase is used here to store the semantic embeddings generated from entities. In this step, we define a method to connect to the Couchbase database using the provided credentials.\n",
+    "\n",
+    "The CouchbaseVectorStore allows you to store, retrieve, and manage vector embeddings in Couchbase.\n",
+    "The connect() method initializes the connection to Couchbase using the provided connection string, username, and password."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "kiYpzj7-gdC4"
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024-09-06 15:34:48,892 - __main__ - INFO - Setting up CouchbaseVectorStore\n",
+      "2024-09-06 15:34:48,898 - graphrag.vector_stores.couchbasedb - INFO - Connecting to Couchbase at couchbase://localhost\n",
+      "2024-09-06 15:34:48,966 - graphrag.vector_stores.couchbasedb - INFO - Successfully connected to Couchbase\n",
+      "2024-09-06 15:34:48,970 - __main__ - INFO - CouchbaseVectorStore setup completed\n"
+     ]
+    }
+   ],
+   "source": [
+    "logger.info(\"Setting up CouchbaseVectorStore\")\n",
+    "\n",
+    "try:\n",
+    "    description_embedding_store = CouchbaseVectorStore(\n",
```

Comment:
Not sure but some other naming may be useful?

Reply:
named it to couchbase vector store